platform/lib/templates/index.js

```diff
@@ -97,8 +118,14 @@ class Templates {
   async fetchTemplate_(templatePath) {
     const templateUrl = new URL(templatePath, config.hosts.pages.base);
     const fetchResponse = await fetch(templateUrl);
-    return fetchResponse.text();
+
+    if (fetchResponse.status && fetchResponse.status !== 404) {
```

Comment:
use `fetchResponse.ok` instead

Reply:
`Response.ok` would be `false` if Grow returned a error debug page which would have a 500 status but you'd like to return to the user. That's why I went for the more verbose status checking.

Reply:
add a comment in that case